tests/unit/javascript/conftest.py

```diff
@@ -68,6 +78,38 @@ def javaScriptConsoleMessage(self, msg, line, source):
             """Fail tests on js console messages as they're used for errors."""
             pytest.fail("js console ({}:{}): {}".format(source, line, msg))
 
+if QWebEnginePage is None:
+    TestWebEnginePage = None
+else:
+    class TestWebEnginePage(QWebEnginePage):
+
+        """QWebPage subclass which overrides some test methods.
+
+        Attributes:
+            _logger: The logger used for alerts.
+        """
+
+        def __init__(self, parent=None):
+            super().__init__(parent)
+            self._logger = logging.getLogger('js-tests')
+
+        def javaScriptAlert(self, _frame, msg):
+            """Log javascript alerts."""
+            self._logger.info("js alert: {}".format(msg))
+
+        def javaScriptConfirm(self, _frame, msg):
+            """Fail tests on js confirm() as that should never happen."""
+            pytest.fail("js confirm: {}".format(msg))
+
+        def javaScriptPrompt(self, _frame, msg, _default):
+            """Fail tests on js prompt() as that should never happen."""
+            pytest.fail("js prompt: {}".format(msg))
+
+        def javaScriptConsoleMessage(self, level, msg, line, source):
+            """Fail tests on js console messages as they're used for errors."""
+            pytest.fail("[{}] js console ({}:{}): {}".format(level, source,
```

Comment:
This gets a `JavaScriptConsoleMessageLevel`, which is a cpp enum. Is there a way to get the cpp enum as a text value for easy printing? I don't even know a clean way of getting the string value of an enum in cpp...

Reply:
Not in C++ in general (it doesn't support that), but qutebrowser has [qenum_key](https://github.com/qutebrowser/qutebrowser/blob/7cc3fb4a4ea7ec4b05129af53b29e44edfe9c276/qutebrowser/utils/debug.py#L95-L131) in `qutebrowser.utils.debug` which tries asking Qt's [meta objects](http://doc.qt.io/qt-5/metaobjects.html), or failing that, checks all possible values to get the name.